src/Sarif/Autogenerated/FileData.cs

```diff
@@ -56,6 +56,12 @@ public SarifNodeKind SarifNodeKind
         [DataMember(Name = "length", IsRequired = false, EmitDefaultValue = false)]
         public int Length { get; set; }
 
+        /// <summary>
+        /// The role or roles played by the file in the analysis.
+        /// </summary>
+        [DataMember(Name = "roles", IsRequired = false, EmitDefaultValue = false)]
+        public IList<string> Roles { get; set; }
```

Comment:
These should be a `FileRoles` enum, by way of an `EnumHint` in CodeGenHints.json.

Reply:
Fixed